Sources/Build/BuildPlan.swift

```diff
@@ -0,0 +1,486 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Basic
+import Utility
+import PackageModel
+import PackageGraph
+import PackageLoading
+import func POSIX.getenv
+
+public struct BuildParameters {
+
+    /// The path to the data directory.
+    public let dataPath: AbsolutePath
+
+    /// The build configuration.
+    public let configuration: Configuration
+
+    /// The path to the build directory (inside the data directory).
+    public var buildPath: AbsolutePath {
+        return dataPath.appending(component: configuration.dirname)
+    }
+
+    /// The toolchain.
+    public let toolchain: Toolchain
+
+    /// Extra build flags.
+    public let flags: BuildFlags
+
+    /// Extra flags to pass to Swift compiler.
+    public var swiftCompilerFlags: [String] {
+        var flags = self.flags.cCompilerFlags.flatMap{ ["-Xcc", $0] }
+        flags += self.flags.swiftCompilerFlags
```

Comment:
Is it always correct to pass the `clang` compiler flags first?  It could be that the `swiftc` front-end just pends all the `clang` flags anyway, so that their order compared with the `swiftc` flags doesn't matter, but I don't know for sure.